lib/services/prepare-platform-native-service.ts

```diff
@@ -62,13 +63,36 @@ export class PreparePlatformNativeService extends PreparePlatformService impleme
 
 	private copyAppResources(platformData: IPlatformData, projectData: IProjectData): void {
 		const appDestinationDirectoryPath = path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME);
-		const appResourcesDirectoryPath = path.join(appDestinationDirectoryPath, constants.APP_RESOURCES_FOLDER_NAME);
-		if (this.$fs.exists(appResourcesDirectoryPath)) {
-			platformData.platformProjectService.prepareAppResources(appResourcesDirectoryPath, projectData);
+		const appResourcesDestinationDirectoryPath = path.join(appDestinationDirectoryPath, constants.APP_RESOURCES_FOLDER_NAME);
+		if (this.$fs.exists(appResourcesDestinationDirectoryPath)) {
+			platformData.platformProjectService.prepareAppResources(appResourcesDestinationDirectoryPath, projectData);
 			const appResourcesDestination = platformData.platformProjectService.getAppResourcesDestinationDirectoryPath(projectData);
 			this.$fs.ensureDirectoryExists(appResourcesDestination);
-			shell.cp("-Rf", path.join(appResourcesDirectoryPath, platformData.normalizedPlatformName, "*"), appResourcesDestination);
-			this.$fs.deleteDirectory(appResourcesDirectoryPath);
+
+			if (platformData.normalizedPlatformName.toLowerCase() === "android") {
+				const appResourcesDirectoryPath = projectData.getAppResourcesDirectoryPath();
+				const appResourcesDirStructureHasMigrated = this.$androidResourcesV4MigrationService.hasMigrated(appResourcesDirectoryPath);
+				const appResourcesAndroid = path.join(appResourcesDirectoryPath, platformData.normalizedPlatformName);
+
+				if (appResourcesDirStructureHasMigrated) {
+					shell.cp("-Rf", path.join(appResourcesAndroid, "src", "main"), appResourcesDestination);
```

Comment:
you can use `this.$fs.copyFile` instead of `shell.cp` (it uses `shelljs.cp('-rf'...` internally). This will make the code easier for testing (unit tests).